src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingSmelterCrafter.java

```diff
@@ -106,6 +106,13 @@ public int getMaxBuildingLevel()
         return builder.build();
     }
 
+    @Override
+    public int buildingRequiresCertainAmountOfItem(final ItemStack stack, final List<ItemStorage> localAlreadyKept, final boolean inventory)
+    {
+        localAlreadyKept.addAll(getAllowedFuel().stream().map(is -> new ItemStorage(is)).filter(i -> i.getItemStack().equals(stack)).collect(Collectors.toCollection(ArrayList::new)));
+        return super.buildingRequiresCertainAmountOfItem(stack, localAlreadyKept, inventory);
+    }
```

Comment:
Why would we store all allowed fuel as already kept?

Reply:
besides being an expensive stream, I don't think this is doing what you think it is. You put in the localAlreadyKept the blocks you did not dump. This way it will dump them. What you want to do is to check if stack is part of allowed fuel and then if localAlreadyKept contains 64 of allowed fuel. If not, then you return the quantity yu want to dump (to make sure at least 64 stay in the workers inv), else you'll return stack.size and if not allowed fuel you return super

Reply:
As you are filtering only on the ItemStack and not on the ItemStorage it would make sense to swap the order of operations to prevent creating unneeded ItemStorage instances:
localAlreadyKept.addAll(getAllowedFuel().stream().filter(i -> i.equals(stack)).map(ItemStorage::new).collect(Collectors.toCollection(ArrayList::new)));

Reply:
This code is gone.